docs/source/traitsui_user_manual/testing.rst

```diff
@@ -0,0 +1,628 @@
+nested GUI elements, or perform a user interaction on the GUI element.
+
+The supported actions can be found via the |UIWrapper.help| method::
+
+    from traitsui.testing.api import UITester
+
+    class Person(HasTraits):
+        name = Str()
+
+    class App(HasTraits):
+        title = Str()
+        person = Instance(Person, ())
+
+    view = View(
+        Item(name="title"),
+        Item(name="person", editor=InstanceEditor(), style="custom"))
+
+    obj = App()
+    tester = UITester()
+    with tester.create_ui(obj, dict(view=view)) as ui:
+        title_field = tester.find_by_name(ui, "title")
+
+        # Print some help message
+        title_field.help()
+
+The last line ``title_field.help()`` prints something like this (abbreviated
+for the purpose of this section)::
+
+    Interactions
+    ------------
+    <class 'traitsui.testing.tester.command.KeyClick'>
+        An object representing the user clicking a key on the keyboard.
+        ...
+
+    <class 'traitsui.testing.tester.command.KeySequence'>
+        An object representing the user typing a sequence of keys.
+        ...
+
+    <class 'traitsui.testing.tester.command.MouseClick'>
+        An object representing the user clicking a mouse button.
+        ...
+
+    <class 'traitsui.testing.tester.query.DisplayedText'>
+        An object representing an interaction to obtain the displayed
+        (echoed) plain text.
+        ...
+
+    Locations
+    ---------
+    No locations are supported.
+
+The "Interactions" section shows the types of objects that can be used with
+|UIWrapper.perform| and |UIWrapper.inspect|. They are objects that
+represent user actions such as clicking a mouse, or checking a text being
+displayed.
+
+The "Locations" section shows the types of objects that can be used with
+|UIWrapper.locate|. They are objects that allow developers to navigate
+further into the the current GUI element.
+
+In this example, ``title_field`` is wrapping a textbox, no further
+nested GUI elements can be located and therefore there are no locations
+supported.
+
+Most of the time these objects can be imported from
+``traitsui.testing.api``.
+
+Perform a user interaction to modify GUI state
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+To modify the GUI state, we use |UIWrapper.perform| with an object whose
+type is supported (as shown in the "Interactions" section from the help
+message).
+
+Say we want to modify the value in the text box as if the user has typed
+in it, we can use the ``KeySequence`` object together with
+|UIWrapper.perform|::
+
+    from traitsui.testing.api import UITester, KeySequence
+
+    obj = App()
+    tester = UITester()
+    with tester.create_ui(obj, dict(view=view)) as ui:
+        title_field = tester.find_by_name(ui, "title")
+        title_field.perform(KeySequence("New Title"))
+        assert obj.title == "New Title"
+
+We can then check the trait being edited is updated (many times,
+once per keystroke).
```

Comment:
I think the "many times, once per keystroke" deserves more explanation. It essentialy means that `enter_set` is `False` so the trait is updated with every key press. Is that correct?